smallrye-reactive-messaging-provider/src/main/java/io/smallrye/reactive/messaging/connectors/WorkerPoolRegistry.java

```diff
@@ -0,0 +1,133 @@
+        Objects.requireNonNull(blockingCodeHandler, "Code to execute not provided");
+
+        if (workerName == null) {
+            return executionHolder.vertx().executeBlocking(blockingCodeHandler, ordered);
+        } else {
+            return getWorker(workerName).executeBlocking(blockingCodeHandler, ordered);
+        }
+    }
+
+    private WorkerExecutor getWorker(String workerName) {
+        Objects.requireNonNull(workerName, "Worker Name not specified");
+
+        if (workerExecutors.containsKey(workerName)) {
+            return workerExecutors.get(workerName);
+        }
+        if (workerDefinitions.containsKey(workerName)) {
+            WorkerExecutor executor = executionHolder.vertx().createSharedWorkerExecutor(workerName,
+                    workerDefinitions.get(workerName));
+            if (executor != null) {
+                workerExecutors.put(workerName, executor);
+                return executor;
+            } else {
+                throw new RuntimeException("Failed to create Worker for " + workerName);
+            }
+        }
+
+        // Shouldn't get here
+        throw new IllegalArgumentException("@Blocking referred to invalid worker name.");
+    }
+
+    public <T> void analyzeWorker(AnnotatedType<T> annotatedType) {
+        Objects.requireNonNull(annotatedType, "AnnotatedType was empty");
+
+        Set<AnnotatedMethod<? super T>> methods = annotatedType.getMethods();
+
+        methods.stream()
+                .filter(m -> m.isAnnotationPresent(Blocking.class))
+                .forEach(m -> addWorker(m.getJavaMember()));
+    }
+
+    private void addWorker(Method method) {
+        Objects.requireNonNull(method, "Method was empty");
+
+        Blocking blocking = method.getAnnotation(Blocking.class);
+
+        // Validate @Blocking is used in conjunction with @Incomings, @Incoming, or @Outgoing
+        if (!(method.isAnnotationPresent(Incomings.class) || method.isAnnotationPresent(Incoming.class)
+                || method.isAnnotationPresent(Outgoing.class))) {
+            throw getBlockingError(method, "no @Incomings, @Incoming, or @Outgoing present");
+        }
+
+        if (!blocking.value().equals(Blocking.NO_VALUE)) {
+            // Validate @Blocking value is not empty, if set
+            if (Validation.isBlank(blocking.value())) {
+                throw getBlockingError(method, "value is blank or null");
+            }
+
+            // Validate @Blocking worker pool has configuration to define concurrency
+            String workerConfigKey = WORKER_CONFIG_PREFIX + "." + blocking.value() + "." + WORKER_CONCURRENCY;
+            Optional<Integer> concurrency = configInstance.get().getOptionalValue(workerConfigKey, Integer.class);
+            if (!concurrency.isPresent()) {
+                throw getBlockingError(method, workerConfigKey + " was not defined");
+            }
+
+            workerDefinitions.put(blocking.value(), concurrency.get());
```

Comment:
Maybe a log message (info) indicating the that the worker pool has been created with concurrency x